src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -123,6 +123,78 @@ public new NullabilityInfo TopLevelNullability
 
         public CodeAnalysis.ITypeSymbol? GetPublicTypeSymbol()
             => Type?.GetITypeSymbol(TopLevelNullability.FlowState.ToAnnotation());
+
+        public virtual bool IsEquivalentToThisReference => false;
+    }
+
+    internal partial class BoundValuePlaceholderBase
+    {
+        public abstract override bool IsEquivalentToThisReference { get; }
+    }
+
+    internal partial class BoundValuePlaceholder
+    {
+        public sealed override bool IsEquivalentToThisReference => throw ExceptionUtilities.Unreachable;
+    }
+
+    internal partial class BoundInterpolatedStringHandlerPlaceholder
+    {
+        public sealed override bool IsEquivalentToThisReference => false;
+    }
+
+    internal partial class BoundDeconstructValuePlaceholder
+    {
+        public sealed override bool IsEquivalentToThisReference => false; // Preserving old behavior
```

Comment:
nit: I found the comment confusing. If the behavior is incorrect, should we file an issue instead?
Same for BoundAwaitableValuePlaceholder

Reply:
>  If the behavior is incorrect, should we file an issue instead?

That is not something that I intend to pursue, given that this only affects warnings. If you think we have to follow up anyway, I can open an issue. Let me know.